miso-web/src/main/webapp/scripts/hot_library.js

```diff
@@ -819,8 +819,34 @@ HotTarget.library = (function() {
           HotUtils.makeParents('library', HotUtils.relationCategoriesForDetailed()),
           HotUtils.makeChildren('library', [HotUtils.relations.dilution(), HotUtils.relations.pool()])].concat(
           HotUtils.makeQcActions("Library")).concat(
-          config.worksetId ? [HotUtils.makeRemoveFromWorkset('libraries', config.worksetId)] : [HotUtils.makeAddToWorkset('libraries',
-              'libraryIds')]);
+          [
+              config.worksetId ? HotUtils.makeRemoveFromWorkset('libraries', config.worksetId) : HotUtils.makeAddToWorkset('libraries',
```

Comment:
I think you can shove this in a function that would have the form:
```
function makeAttachFile('library', function(library) { return library.parentSampleProjectId; })
```
The rest seems to be the same as the samples version.

Reply:
done